src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameNodeDefinitions.cs

```diff
@@ -0,0 +1,426 @@
+            if (index == ChildCount - 1)
+            {
+                return CloseToken;
+            }
+
+            // Includes both the nodes and separator tokens as children
+            return TypeArguments.NodesAndTokens[index - 1];
+        }
+    }
+
+    internal sealed class StackFrameParameterDeclarationNode : StackFrameDeclarationNode
+    {
+        public readonly StackFrameNameNode Type;
+        public readonly StackFrameToken Identifier;
+
+        internal override int ChildCount => 2;
+
+        public StackFrameParameterDeclarationNode(StackFrameNameNode type, StackFrameToken identifier)
+            : base(StackFrameKind.Parameter)
+        {
+            Debug.Assert(identifier.Kind == StackFrameKind.IdentifierToken);
+            Type = type;
+            Identifier = identifier;
+        }
+
+        public override void Accept(IStackFrameNodeVisitor visitor)
+            => visitor.Visit(this);
+
+        internal override StackFrameNodeOrToken ChildAt(int index)
+            => index switch
+            {
+                0 => Type,
+                1 => Identifier,
+                _ => throw new InvalidOperationException()
+            };
+    }
+
+    internal sealed class StackFrameParameterList : StackFrameNode
+    {
+        public readonly StackFrameToken OpenParen;
+        public readonly EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameParameterDeclarationNode> Parameters;
+        public readonly StackFrameToken CloseParen;
+
+        public StackFrameParameterList(
+            StackFrameToken openToken,
+            EmbeddedSeparatedSyntaxNodeList<StackFrameKind, StackFrameNode, StackFrameParameterDeclarationNode> parameters,
+            StackFrameToken closeToken)
+            : base(StackFrameKind.ParameterList)
+        {
+            Debug.Assert(openToken.Kind == StackFrameKind.OpenParenToken);
+            Debug.Assert(closeToken.Kind == StackFrameKind.CloseParenToken);
+
+            OpenParen = openToken;
+            Parameters = parameters;
+            CloseParen = closeToken;
+        }
+
+        internal override int ChildCount => 2 + Parameters.NodesAndTokens.Length;
+
+        public override void Accept(IStackFrameNodeVisitor visitor)
+        {
+            visitor.Visit(this);
+        }
+
+        internal override StackFrameNodeOrToken ChildAt(int index)
+        {
+            if (index == 0)
+            {
+                return OpenParen;
+            }
+
+            if (index == ChildCount - 1)
+            {
+                return CloseParen;
+            }
+
+            // Include both nodes and tokens here as children of the StackFrameParameterList
+            return Parameters.NodesAndTokens[index - 1];
+        }
+    }
+
+    internal sealed class StackFrameFileInformationNode : StackFrameNode
+    {
+        public readonly StackFrameToken Path;
+        public readonly StackFrameToken? Colon;
+        public readonly StackFrameToken? Line;
```

Comment:
in gneeral, we don't model with nullable tokens, we model with tokens whose kind is .none.  but if this makes this simpler here, then that's ok.

Reply:
I think this tends to be cleaner here because it lets the caller know they should look for `null`. Without this caller would have to have intrinsic knowledge that `None` was a possible kind on the token without any indication in the type.

Reply:
It may be a moot point, but in my head I like this better. I think as I start to use this in the UI layer I may change my mind